scripts/vale-review.sh

```diff
@@ -15,90 +15,55 @@ else
+    curl -L -X POST -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repos/openshift/openshift-docs/pulls/$PULL_NUMBER/comments -d '{"body":"'"$BODY"'","commit_id":"'"$COMMIT_ID"'","path":"'"$FILENAME"'","line":'"$LINE_NUMBER"',"side":"RIGHT"}'
 
 }
 
-# Get updated files in the last commit.
-FILES=$(git diff --name-only HEAD~1 HEAD --diff-filter=d "*.adoc" ':(exclude)_unused_topics/*')
-
-# Compares body, path, and line values for Vale alerts vs already existing comments on PR. If values match, comment already exists, remove from temp json files.
-function check_if_existing_comment {
-
-    echo "Checking if existing comment..."
-
-    local vale_file="$1"
-    local pull_comments_file="$2"
-
-    # Iterate through Vale alert JSON
-    jq -c '.[]' "$vale_file" | while IFS= read -r object; do
-    BODY_VALE=$(echo "$object" | jq -r '.body')
-    FILENAME_VALE=$(echo "$object" | jq -r '.path')
-    LINE_NUMBER_VALE=$(echo "$object" | jq -r '.line')
-
-        # Iterate through pull comment JSON to see if a comment exists
-        jq -c '.[]' "$pull_comments_file" | while IFS= read -r object2; do
-            BODY_COMMENT=$(echo "$object2" | jq -r '.body')
-            FILENAME_COMMENT=$(echo "$object2" | jq -r '.path')
-            LINE_NUMBER_COMMENT=$(echo "$object2" | jq -r '.line')
-
-            # If a there's a comment that matches the Vale body, line and path, remove it from both files
-            if [ "$BODY_VALE" == "$BODY_COMMENT" ] && [ "$FILENAME_VALE" == "$FILENAME_COMMENT" ] && [ "$LINE_NUMBER_VALE" == "$LINE_NUMBER_COMMENT" ]; then
-                echo "Found existing comment for $object"
-                echo "Removing alert because review comment already exists... "
-                
-                jq --argjson obj "$object" '. | del(.[index($obj)])' "$vale_file" > processing.json
-                jq --argjson obj "$object2" '. | del(.[index($obj)])' "$pull_comments_file" > processing_comments.json
-
-                mv processing.json "$vale_file"
-                mv processing_comments.json "$pull_comments_file"
+function get_vale_errors {
+    echo "Getting the Vale errors and PR comments and filtering out existing comments..."
+    local vale_json="$1"
+    local pull_comments_json="$2"
 
-                break
-
-            fi
-        done
-
-    done
+    # jq map and filter for new review comments only
+    updated_comments_json=$(jq -n --argjson vale "$vale_json" --argjson comments "$pull_comments_json" '$vale | map(select(. as $v | $comments | any(.path == $v.path and .line == $v.line) | not))' | jq)
 
+    export updated_comments_json
```

Comment:
If I echo updated_comments_json I get the following:

````
[
    {
        "body": "🤖 adding_service_cluster architecture backup_and_restore cli_reference Dockerfile Gemfile Guardfile Jenkinsfile Makefile microshift_backup_and_restore microshift_storage microshift_welcome mod_docs_guide osd_architecture osd_install_access_delete_cluster Rakefile rosa_architecture router scalability_and_performance storage web_console welcome RedHat.CaseSensitiveTerms**: Use 'Microsoft Azure' rather than 'Azure'.",
        "path": "welcome/index.adoc",
        "line": 84
    },
    {
        "body": "🤖 adding_service_cluster architecture backup_and_restore cli_reference Dockerfile Gemfile Guardfile Jenkinsfile Makefile microshift_backup_and_restore microshift_storage microshift_welcome mod_docs_guide osd_architecture osd_install_access_delete_cluster Rakefile rosa_architecture router scalability_and_performance storage web_console welcome RedHat.CaseSensitiveTerms**: Use 'Microsoft Azure' rather than 'Azure'.",
        "path": "welcome/index.adoc",
        "line": 86
    },
    {
        "body": "🤖 adding_service_cluster architecture backup_and_restore cli_reference Dockerfile Gemfile Guardfile Jenkinsfile Makefile microshift_backup_and_restore microshift_storage microshift_welcome mod_docs_guide osd_architecture osd_install_access_delete_cluster Rakefile rosa_architecture router scalability_and_performance storage web_console welcome RedHat.TermsErrors**: Use 'user-provisioned infrastructure' rather than 'upi'.",
        "path": "welcome/index.adoc",
        "line": 127
    },
    {
        "body": "🤖 adding_service_cluster architecture backup_and_restore cli_reference Dockerfile Gemfile Guardfile Jenkinsfile Makefile microshift_backup_and_restore microshift_storage microshift_welcome mod_docs_guide osd_architecture osd_install_access_delete_cluster Rakefile rosa_architecture router scalability_and_performance storage web_console welcome RedHat.CaseSensitiveTerms**: Use 'Microsoft Azure' rather than 'Azure'.",
        "path": "welcome/index.adoc",
        "line": 157
    },
    {
        "body": "🤖 adding_service_cluster architecture backup_and_restore cli_reference Dockerfile Gemfile Guardfile Jenkinsfile Makefile microshift_backup_and_restore microshift_storage microshift_welcome mod_docs_guide osd_architecture osd_install_access_delete_cluster Rakefile rosa_architecture router scalability_and_performance storage web_console welcome RedHat.TermsErrors**: Use 'before' rather than 'prior to'.",
        "path": "welcome/index.adoc",
        "line": 247
    }
]
````

Reply:
nit - I find it more understandable to call this variable `updated_vale_json`. The JSON contains the remaining Vale errors that don't have an existing PR comment.  So it's not really "comments_json".... in my head at least :-)

Reply:
> If I echo updated_comments_json I get the following

I can't replicate this. Are you hot-wiring the script somehow? Checkout the PR and try it.

Reply:
Updated to "updated_vale_json"